packages/devtools/lib/src/timeline/timeline_service.dart

```diff
@@ -28,7 +29,7 @@ class TimelineService {
   }
 
   void _handleConnectionStart(VmServiceWrapper service) {
-    serviceManager.service.setFlag('profile_period', '250');
+    serviceManager.service.setFlag('profile_period', '$defaultSamplePeriod');
```

Comment:
If you have multiple instances of devtools open, the sample period value could get out of sync between them.
The VM does not appear to yet dispatch an event when a flag value changes so it wouldn't be easy to fully keep the values in sync today. Instead we should file a bug to get such an event added to the VM.
What we could do today is on startup read the existing flag value from the VM and use that value if it is not the default (the VM will tell you). That way a second instance of DevTools wouldn't trample over the value of the flag set by a different instance of DevTools.

Reply:
The only issue I see with this is what the default value is. I don't think it is a good idea to default to 1000 (which is the VM default), because then a user is going to get minimal cpu profiling data and they may not know that they can change it. Playing with some numbers yesterday, it seemed like 250 was a good default value to start with.

Is there a way to observe when a chrome tab is active? That could also work because when the chrome tab becomes active, we can set the profile_period to whatever is showing on the page's selector.

I'm not convinced profile_period is something we necessarily want to keep in sync across tabs and pages? I can see a use case where you'd want different values for timeline vs performance.

Reply:
unfortunately it is something we mainly have to keep in sync as a user might have multiple tabs open at the same time.
https://github.com/dart-lang/sdk/blob/master/runtime/vm/service/service.md#flag
has a boolean to indicate whether it was modified so we can tell if the setting was the VM default or not.